backup/re_initialise.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+#
+# (Re-)initialise a system from a backup that has been dumped into the root tree.
+
+chown mastodon:mastodon /opt/mastodon/mastodon.env
+
+source /opt/mastodon/versions.env
+chmod 0444 /opt/backup/mastodon_db_backup.sql
+
+# We start the DB container to import the backup SQL dump.
+# Note that the container's health check will not succeed until after the dump was imported.
+echo "Importing mastodon database"
+# We run this in the background instead of using --detach so the log output remains visible
+/usr/bin/docker run --rm  \
+                  --shm-size=256mb \
+                  --health-cmd="pg_isready -U postgres" \
+                  --env POSTGRES_HOST_AUTH_METHOD=trust \
+                  --volume=/opt/backup/mastodon_db_backup.sql:/docker-entrypoint-initdb.d/mastodon_db.sql \
+                  --volume=/opt/mastodon/postgres:/var/lib/postgresql/data \
+                  --name=mastodon-db-import \
+                  "${POSTGRES_IMAGE}" &
+
+# 20 minutes max for db import. The DB container can be shut down as soon as the DB is up (healthy)
+# as the SQL import happens before the DB is started in the container.
+/opt/mastodon/wait-for-container.sh mastodon-db-import .State.Health healthy 1200
+/usr/bin/docker stop mastodon-db-import
+
+rm /opt/backup/mastodon_db_backup.sql
+
+echo "Enabling mastodon and monitoring service units"
+systemctl daemon-reload
+
+for f in /etc/systemd/system/mastodon* \
+         /etc/systemd/system/metrics-provider* \
+         /etc/systemd/system/monitoring* \
+         caddy; do
+
+         if ! grep -q '\[Install\]' "${f}"; then
+            # Some services are pulled in by other services and do not need installation.
+            continue
+         fi
```

Comment:
Now I noticed that `caddy` is not a path, so I'm wondering if enabling caddy won't be skipped, because grep will likely talk about `caddy` file not being found. Maybe the `caddy` line could also be a path?